projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/SupportedPythonVersions.java

```diff
@@ -0,0 +1,131 @@
+   */
+  public List<String> getSupportedPythonVersions() {
+    if (supportedPythonVersions != null && !supportedPythonVersions.isEmpty()) {
+      return supportedPythonVersions.keySet().stream()
+          .map((obj) -> Objects.toString(obj, null))
+          .collect(Collectors.toList());
+    } else {
+      return Arrays.asList(getDefaultPythonVersion());
+    }
+  }
+
+  /**
+   * Returns the name of the data job base image as stored in the docker registry. If
+   * supportedPythonVersions is set, and the pythonVersion passed by the user is supported according
+   * to the configuration, the base image corresponding to the pythonVersion is returned. Otherwise,
+   * the default base image is returned.
+   *
+   * @param pythonVersion a string indicating the python version passed by the user
+   * @return a string of the data job base image.
+   */
+  public String getJobBaseImage(String pythonVersion) {
+    if (!supportedPythonVersions.isEmpty() && isPythonVersionSupported(pythonVersion)) {
+      return supportedPythonVersions.get(pythonVersion).get(BASE_IMAGE);
+    } else {
+      log.warn(
+          "An issue with the passed pythonVersion or supportedPythonVersions configuration has"
+              + " occurred. Returning default job base image");
+      return getDefaultJobBaseImage();
+    }
+  }
+
+  public String getDefaultJobBaseImage() {
+    return supportedPythonVersions.get(defaultPythonVersion).get(BASE_IMAGE);
+  }
+
+  /**
+   * Returns the name of the vdk image as stored in the docker registry. If supportedPythonVersions
+   * is set, and the pythonVersion, passed by the user, is supported according to the configuration,
+   * the vdk image corresponding to the pythonVersion is returned. Otherwise, the default vdk image
+   * is returned.
+   *
+   * @param pythonVersion a string indicating the python version passed by the user
+   * @return a string of the vdk image.
+   */
+  public String getVdkImage(String pythonVersion) {
+    if (!supportedPythonVersions.isEmpty() && isPythonVersionSupported(pythonVersion)) {
+      return supportedPythonVersions.get(pythonVersion).get(VDK_IMAGE);
+    } else {
+      log.warn(
+          "An issue with the passed pythonVersion or supportedPythonVersions configuration has"
+              + " occurred. Returning default vdk image");
+      return getDefaultVdkImage();
+    }
+  }
+
+  public String getDefaultVdkImage() {
+    return supportedPythonVersions.get(defaultPythonVersion).get(VDK_IMAGE);
+  }
+
+  /**
+   * Returns the default python version supported by the Control Service. The version number is read
+   * from the datajobs.deployment.defaultPythonVersion application property.
+   *
+   * @return a string indicating the default python version supported by the Control Service.
+   * @throws InvalidParameterException
+   */
+  public String getDefaultPythonVersion() throws InvalidParameterException {
```

Comment:
not needed any more

Reply:
I am not sure I understand. We need a method to return the default version to the users.